### ISSUE TITLE
[QUESTION]an error of creating the run directory in GC 13.4.1

### ISSUE THREAD

**Issue by zhiweistu** [geoschem/geos-chem#1307]
### Ask a question about GEOS-Chem:
Hi, everyone. 
I am quite new to GC. My GC is 13.4.1. I try creating the run directory, choosing the fullchem, standard simulation, but there is an error. 

![image](https://user-images.githubusercontent.com/109276335/179213407-16cab2a2-4b3e-4ba2-aaa5-130f74748ebf.png)

I think the error may stem from missing restart files，but I set “ExtData” as the data read path, and GC restart file is in GEOSCHEM_RESTARTS(“ExtData” subdirectory).

![image](https://user-images.githubusercontent.com/109276335/179212039-6cabec96-5acb-473a-8d3d-a1433df258ca.png)
![image](https://user-images.githubusercontent.com/109276335/179212083-4c5bc766-6881-48ea-aa38-da24147a08c5.png)

Looking forward for your helps, thanks!

**Comment by lizziel** [comment]
Hi @zhiweistu, try finding the following lines in `createRunDir.sh` which you are using create the run directory:

```
    printf "\n  -- No sample restart provided for this simulation."
    printf "\n     You will need to provide an initial restart file or disable"
    printf "\n     GC_RESTARTS in HEMCO_Config.rc to initialize your simulation"
    printf "\n     with default background species concentrations.\n"
```

Then add this line to that block using the same indentation, and rerun. This should print out what restart file is being looked for, which might help figure out the problem. I suspect there may be something wrong with the path.

`    printf "${sample_rst}"`

**Comment by zhiweistu** [comment]
Hi@lizziel, thanks for your help. As you said, the “restart file error” is path wrong. I added this line in block, and I found it needed restart file”GEOSChem.Restart.fullchem.20190701_0000z.nc4 ”, and the path was .../GEOSCHEM_RESTARTS/v2021-09. 
![image](https://user-images.githubusercontent.com/109276335/179253827-22e3899a-154b-41a6-adf8-dd932e5345e0.png)

The v2021-09 did not exist before(I am curious about why it choose that path), so I create it, and added the restart file in the directory. The error about restart file disappeared. 
But there are still some errors about “can not read History.rc”, error about “HEAD”. Please give me some helps again, thanks for your helps.
![image](https://user-images.githubusercontent.com/109276335/179254013-9f33d880-e01d-4a6a-b76d-ca29b3ad6f4a.png)

**Comment by lizziel** [comment]
Hi @zhiweistu, I am glad that helped. I failed to notice before the first error message of `realpath: command not found`. That is happening when calling `init_rd.sh`, a helper script used to make substitutions within our config file templates. It's possible your system does not have the `realpath` function available. Open `init_rd.sh` and try using `readlink -f` in place of `realpath` and see if that fixes the rest of the problems.

Regarding the restart files, did you download ExtData a while ago? If yes, you should sync to the latest data so that you have everything you need to run the version you are using. We have a tool called `BashDataCatalog` that you can use to do this. For documentation see https://geos-chem.readthedocs.io/en/latest/geos-chem-shared-docs/supplemental-guides/bashdatacatalog.html.

**Comment by zhiweistu** [comment]
Hi@lizziel, I am so grateful for your help. I used readlink -f in place of real path, it successfully solved the error about read file. 
But the last error about “HEAD” still existed. Some helps please, thanks!
![image](https://user-images.githubusercontent.com/109276335/179340089-3e8dd421-62d2-40e6-a6cf-fe8117c779bb.png)

**Comment by lizziel** [comment]
Hi @zhiweistu, what version of git are you using? I think the error is occurring in this line:
`code_branch=$(git rev-parse --abbrev-ref HEAD)`

Try commenting out that line along with the line further down that uses `${code_branch}`. Also add the following print after `cd ${gc_dir}` (two lines before setting `code_branch`) to make sure the path to GEOS-Chem is correct. 
`printf "GEOS-Chem repository path: ${gcdir}"`

This code is supposed to get the version information of the GEOS-Chem repository to print to run directory creation log file `rundir.version`. If that is not working then I think it is either due to a very old git version or a problem getting to the GEOS-Chem directory before running the git commands.

**Comment by zhiweistu** [comment]
Hi@lizziel, thanks for the detailed instructions. 
My git version is 1.7.1.
I made 3 code changes according to your instructions.
![image](https://user-images.githubusercontent.com/109276335/179560454-6f3abce6-6894-44b8-8e6c-138ee5b1ae35.png)
                                                                          (modified code )

It printed the “GEOS-Chem repository path: /vol-th/home/hym/GCClassic/src/GEOS-Chem”, the path is right. 
![image](https://user-images.githubusercontent.com/109276335/179560586-c5aa20aa-5a25-49d4-8629-b5b7488697e2.png)
                                                                    (Geos-chem repository path)

Commenting out the code, the information about error “fatal: ambiguous argument ‘HEAD’ : unknown revision or path not in the working tree” disappeared.
But the error “fatal : bad default revision ‘HEAD’ ” still existed. 
So, dose it mean the error stem from the version of git?  
Best Regards
![image](https://user-images.githubusercontent.com/109276335/179560983-840578ef-b313-42f9-8d29-2484f22b2a2a.png)
                                                                                 (run result)

**Comment by lizziel** [comment]
I think the problem is likely your git. Version 1.7.1 is very old, released in 2010, and 1.7.12 is the latest patch for it (meaning you are missing 11 patches with bug fixes following 1.7.11. For comparison, here at Harvard we use 2.17. Try updating a more a recent git and see if that solves the remaining issues.

**Comment by zhiweistu** [comment]
Hi@lizziel, thanks for the detailed instructions. It successfully solved the problem I had before.
But when I compile the file, it always remind me that ifort: command line remark #10120: overriding '-O3' with '-O2'.
 I am not sure if there is an error, my compiler is ifort v13.0, thanks for your help!
Best Regards
![1658664278973](https://user-images.githubusercontent.com/109276335/180646151-d5f61878-2e6e-467d-98b1-8910f009a64c.png)
![1658664173041](https://user-images.githubusercontent.com/109276335/180646094-bf9f43ed-9010-401c-b656-88a474efdbc6.png)

**Comment by lizziel** [comment]
Hi @zhiweistu, I am glad that fixed the problem. Regarding the compiler remark, I suggest switching to a newer compiler rather than do a deep dive into what the issue is. ifort 13 is very old and we no longer test with it. You can try a newer version of ifort, or use gfortran.

**Comment by zhiweistu** [comment]
Hi@lizziel, thanks for your help. I used ifort v18, and the previous error disappeared.
I am sorry to bother you again. When I run the executable gcclassic, The following error message appeared.
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

In Ncrd_3d_R4 #2:  NetCDF: HDF error
     65536         1

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

Code stopped from DO_ERR_OUT (in module NcdfUtil/m_do_err_out.F90) 

This is an error that was encountered in one of the netCDF I/O modules,
which indicates an error in writing to or reading from a netCDF file!

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

I noticed it could not recognize the file format, how can I resolve it?
![image](https://user-images.githubusercontent.com/109276335/182013865-4c59b8dd-4d22-4df8-8d07-56c4f7ece748.png)
                                                                               (13.4.1)

Supplementary information
I have been using gc 12.6.0 before, and its defined diagnostic collections is normal. The format file is CFIO.
My netcdf version is 4.3.2, netcdf - fortran version is 4.2, hdf5 version is 1.8.9. Because  I can used the gc 12.6.0 normally, so I think the netcdf library should be normal.
I am so grateful for your helps !
![image](https://user-images.githubusercontent.com/109276335/182013912-2d2a6410-10f4-46c2-b256-3ec89fc6f522.png)
                                                                               (12.6.0)
